allennlp/tests/data/dataset_readers/dataset_utils/text2sql_utils_test.py

```diff
@@ -0,0 +1,93 @@
+# pylint: disable=no-self-use,invalid-name,line-too-long
+
+import json
+from allennlp.common.testing import AllenNlpTestCase
+from allennlp.data.dataset_readers.dataset_utils import text2sql_utils
+
+class Text2SqlUtilsTest(AllenNlpTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.data = AllenNlpTestCase.FIXTURES_ROOT / 'data' / 'text2sql' / 'restaurants_tiny.json'
+
+    def test_process_sql_data_blob(self):
+
+        data = json.load(open(str(self.data)))
+        dataset = text2sql_utils.process_sql_data_blob(data[0], cross_validation_split=1)
+
+        # All of these data points are the same. This is weird, but currently correct.
+        for split, sql_data in dataset:
+            # This should test because in the data, the cross validation split is == 1.
+            assert split == "test"
+            assert sql_data.text == ['how', 'many', 'buttercup', 'kitchen', 'are', 'there', 'in', 'san', 'francisco', '?']
+            assert sql_data.text_with_variables == ['how', 'many', 'name0', 'are', 'there', 'in', 'city_name0', '?']
+            assert sql_data.sql == ['SELECT', 'COUNT', '(', '*', ')', 'FROM', 'LOCATION', 'AS', 'LOCATIONalias0', ',',
+                                    'RESTAURANT', 'AS', 'RESTAURANTalias0', 'WHERE', 'LOCATIONalias0.CITY_NAME', '=',
+                                    'city_name0', 'AND', 'RESTAURANTalias0.ID', '=', 'LOCATIONalias0.RESTAURANT_ID',
+                                    'AND', 'RESTAURANTalias0.NAME', '=', 'name0', ';']
+            assert sql_data.text_variables == {'city_name0': 'san francisco', 'name0': 'buttercup kitchen'}
+            assert sql_data.sql_variables == {'city_name0': 'san francisco', 'name0': 'buttercup kitchen'}
```

Comment:
Just curious: are the text variables and the sql variables ever different?

Reply:
good question - this only appears in one of the datasets, "advising", which is extremely difficult (bordering on impossible). 
More detail here:
https://github.com/jkkummerfeld/text2sql-data/issues/12